File: src/main/java/xyz/capybara/clamav/commands/Reload.java
package xyz.capybara.clamav.commands;

import xyz.capybara.clamav.exceptions.InvalidResponseException;
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class Reload extends Command<Void> {

    public static final String COMMAND = "RELOAD";

    @Override
    public String getCommandString() {
        return COMMAND;
    }

    @Override
    protected CommandFormat getFormat() {
        return CommandFormat.NULL_CHAR;
    }

    @Override
    protected Void parseResponse(String responseString) {
        if (!responseString.equals("RELOADING")) {
            throw new InvalidResponseException(responseString);
        }

        log.info("Reloading the virus databases");

        return null;
    }
}

File: src/main/kotlin/xyz/capybara/clamav/commands/Reload.kt
package xyz.capybara.clamav.commands

import xyz.capybara.clamav.InvalidResponseException

internal object Reload : Command<Unit>() {
    override val commandString
        get() = "RELOAD"

    override val format
        get() = Command.CommandFormat.NULL_CHAR

    override fun parseResponse(responseString: String) {
        if (responseString != "RELOADING") {
            throw InvalidResponseException(responseString)
        }
        logger.info { "Reloading the virus databases" }
    }
}
